File: src/main/java/com/github/nsc/de/compiler/parser/node/VariableAssignmentNode.java
package com.github.nsc.de.compiler.parser.node;

public class VariableAssignmentNode implements ValuedNode {

    private final String name;
    private final Node value;

    public VariableAssignmentNode(String name, Node value) {
        this.name = name;
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public Node getValue() {
        return value;
    }

    @Override
    public String toString() {
        return this.getName() + " = " + this.getValue();
    }
}

File: shake/compiler/parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/variables/VariableAssignmentNode.kt
package io.github.shakelang.shake.parser.node.variables

import io.github.shakelang.shake.parser.node.*
import io.github.shakelang.parseutils.characters.position.PositionMap

class VariableAssignmentNode(map: PositionMap, val variable: ValuedNode, val value: ValuedNode, val operatorPosition: Int) :
    ValuedNode(map) {

    override fun toJson(): Map<String, *> = mapOf(
        "name" to "VariableAssignmentNode",
        "variable" to variable.json,
        "value" to value.json
    )
}